tests/gtest/avifrgbtoyuvtest.cc

```diff
@@ -839,5 +839,62 @@ INSTANTIATE_TEST_SUITE_P(
                    AVIF_CHROMA_UPSAMPLING_BILINEAR),
             /*has_alpha=*/Bool()));
 
+class SetGetRGBATest
+    : public testing::TestWithParam<std::tuple<
+          /*rgb_depth=*/int, avifRGBFormat, /*is_float=*/bool>> {};
+
+TEST_P(SetGetRGBATest, SetGetTest) {
+  const int rgb_depth = std::get<0>(GetParam());
+  const avifRGBFormat rgb_format = std::get<1>(GetParam());
+  const bool is_float = std::get<2>(GetParam());
+
+  // Unused yuv image, simply needed to initialize the rgb image.
+  testutil::AvifImagePtr yuv(avifImageCreate(4, 2, 8, AVIF_PIXEL_FORMAT_YUV444),
+                             avifImageDestroy);
+
+  testutil::AvifRgbImage rgb(yuv.get(), rgb_depth, rgb_format);
+  rgb.isFloat = is_float;
+
+  avifRGBColorSpaceInfo color_space;
+  if (!avifGetRGBColorSpaceInfo(&rgb, &color_space)) {
+    return;  // Invalid combination e.g. RGB565 with depth > 8.
+  }
+
+  float kEpsilon = 1.0f / color_space.maxChannelF;
+  if (rgb_format == AVIF_RGB_FORMAT_RGB_565) {
+    // Only 5 bits of information per channel except G which has 6).
+    kEpsilon = 1.0f / (1 << 5);
+  }
+
+  const float pixel_to_write[4] = {0.0f, 0.5f, 1.0f, 0.75f};
```

Comment:
These values may be "too easy" for precision round-trip checking. Consider `FillImageGradient()` on a bigger input image with prime dimensions to generate a somewhat better covering distribution.

Reply:
Done with some home made values since FillImageGradient() is for a yuv image (I could convert to rgb but it adds a layer of indirection to the values)

Reply:
There could be a `void FillImageGradient(avifRGBImage* image);` in `aviftest_helpers.h` if used in multiple places. For example `avifalphapremtest.cc` uses `FillImageGradient()` followed by `avifImageYUVToRGB()`. This is fine for now.

Reply:
In this case I'm not even filling an avifRGBImage, I'm filling in a float[4]. I think it's different, and any additional indirection would make it less clear what the test is testing. Also note avifRGBImage doesn't support floats but only half flaots.

Reply:
Right, I got confused with my other comment.